src/Compilers/CSharp/Portable/Parser/DocumentationCommentParser.cs

```diff
@@ -244,7 +244,7 @@ private XmlNodeSyntax ParseXmlElement()
                                 endName = this.WithXmlParseError(endName, XmlParseErrorCode.XML_InvalidWhitespace);
                             }
 
-                            if (!endName.IsMissing && name.ToString() != endName.ToString())
+                            if (!endName.IsMissing && !MatchingXmlNames(name, endName))
```

Comment:
It is not clear why this change is appropriate. It is not semantically equivalent. Do we have new tests covering the difference in behavior?